File: rental-backend/src/main/java/com/realestate/rental/dto/ReportDto.java
package com.realestate.rental.dto;

import java.time.LocalDateTime;
import java.util.UUID;

public record ReportDto(
        UUID id,
        UUID propertyId,
        String propertyTitle,
        UserDto reporter,
        String reason,
        String description,
        String status,
        LocalDateTime createdAt,
        LocalDateTime resolvedAt,
        UserDto resolvedBy
) {}

File: rental-app-flutter/lib/features/admin/models/report_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import '../../auth/models/user_dto.dart';

part 'report_dto.freezed.dart';
part 'report_dto.g.dart';

@freezed
abstract class ReportDto with _$ReportDto {
  const factory ReportDto({
    required String id,
    UserDto? reporter,
    String? propertyId,
    String? reason,
    String? description,
    required String status,
    String? resolvedAt,
    required String createdAt,
  }) = _ReportDto;

  factory ReportDto.fromJson(Map<String, dynamic> json) => _$ReportDtoFromJson(json);
}
